### PR TITLE
Test: Write node ports file before starting tribe service

### PR DESCRIPTION
**Pull request by rjernst**
The tribe service can take a while to initialize, depending on how many
cluster it needs to connect to. This change moves writing the ports file
used by tests to before the tribe service is started.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java [hunk 1]
```diff
@@ -345,7 +344,7 @@ public void apply(Settings value, Settings current, Settings previous) {
                     SearchService.DEFAULT_KEEPALIVE_SETTING,
                     SearchService.KEEPALIVE_INTERVAL_SETTING,
                     SearchService.LOW_LEVEL_CANCELLATION_SETTING,
-                    Node.WRITE_PORTS_FIELD_SETTING,
+                    Node.WRITE_PORTS_FILE_SETTING,
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java]: 👍

**Review by nik9000 — COMMENTED** [review]
Does this mean that the ports file will be written before the node is properly ready? Would that cause spurious test failures?

**Review by ywelsch — APPROVED** [review]
LGTM. Thanks @rjernst

**Comment by rjernst** [comment]
@nik9000 No more so than normal. The node is ready "enough" that the test can poll it to see when the node status is yellow.

**Review by nik9000 — APPROVED** [review]

**Comment by ywelsch** [comment]
The number of nodes in the waitCondition need to be corrected on the tests that use tribe nodes. For example, `smoke-test-tribe-node/build.gradle` would need `ant.get(src: "http://${node.httpUri()}/_cluster/health?wait_for_nodes=>=${numNodes + configOne.numNodes + configTwo.numNodes}&wait_for_status=yellow"`

**Comment by rjernst** [comment]
Thanks for merging @ywelsch!

**Pull request merged**